torch/csrc/api/src/nn/modules/transformer.cpp

```diff
@@ -80,78 +85,58 @@ TransformerDecoderLayerImpl::TransformerDecoderLayerImpl(
 }
 
 void TransformerDecoderLayerImpl::reset() {
-  ///initialize self attention
-  self_attn = register_module(
-    "self_attn",
-    MultiheadAttention(
-      MultiheadAttentionOptions(options.d_model(), options.nhead())
-      .dropout(options.dropout())));
-
-  ///initialize Dropout, post self attention
-  dropout1 = register_module("dropout1",
-    Dropout(DropoutOptions().p(options.dropout())));
-
-  ///initialize Normalization, post self attention
-  norm1 = register_module(
-    "norm1",
-    LayerNorm(LayerNormOptions(std::vector<int64_t> {options.d_model()})));
-
-  ///initialize multihed attention
-  multihead_attn = register_module(
-    "multihead_attn",
-    MultiheadAttention(
-      MultiheadAttentionOptions(options.d_model(), options.nhead())
-      .dropout(options.dropout())));
-
-  ///initialize post multi-headed attention dropout layer
-  dropout2 = register_module(
-    "dropout2", Dropout(DropoutOptions().p(options.dropout())));
-
-  ///initialize post multi-headed attention Normalization
-  norm2 = register_module(
-    "norm2", LayerNorm(
-      LayerNormOptions(std::vector<int64_t> {options.d_model()})));
-
-  ///Initialize Feed forward first linear layer
-  linear1 = register_module(
-    "linear1",
-    Linear(LinearOptions(options.d_model(), options.dim_feedforward())));
-
-  ///initialize Feed forward dropout layer
-  dropout = register_module(
-    "dropout",
-    Dropout(DropoutOptions().p(options.dropout())));
-
-  ///initialize Feed forward second linear layer
-  linear2 = register_module(
-    "linear2",
-    Linear(LinearOptions(options.dim_feedforward(), options.d_model())));
-
-  ///initialize dropout, post feed forward
-  dropout3 = register_module(
-    "dropout3",
-    Dropout(DropoutOptions().p(options.dropout())));
-
-  ///initialize normalization, post feed forward
-  norm3 = register_module(
-    "norm3",
-    LayerNorm(LayerNormOptions(std::vector<int64_t> {options.d_model()})));
+  // NOTE: reset() is for initializing the model only, calling reset() after the model is created
+  // will cause throwing exceptions. Call reset_parameter() if the created model needs a reset.
```

Comment:
nit: "will cause throwing exceptions" -> "will cause exceptions to be thrown", or "will throw exceptions"